Python/Module6_Testing/Hypothesis.md

```diff
@@ -0,0 +1,535 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.3.0
+  kernelspec:
+    display_name: Python [conda env:.conda-jupy] *
+    language: python
+    name: conda-env-.conda-jupy-py
+---
+
+<!-- #raw raw_mimetype="text/restructuredtext" -->
+.. meta::
+   :description: Topic: Writing tests for your code, Difficulty: Easy, Category: Section
+   :keywords: test, automated, pytest, parametrize, fixture, suite, decorator, clean directory  
+<!-- #endraw -->
+
+<!-- #region -->
+# Introduction to Testing with Hypothesis
+
+It is often the case that the process of *describing our data* is by far the heaviest burden that we must bear when writing tests. This process of assessing "what variety of values should I test?", "have I thought of all the important edge-cases?", and "how much is 'enough'?" will crop up with nearly every test that we write.
+Indeed, these are questions that you may have been asking yourself when writing `test_count_vowels_basic` and `test_merge_max_mappings` in the previous sections of this module.
+
+[Hypothesis](https://hypothesis.readthedocs.io/) is a powerful Python library that empowers us to write a _description_ (specification, to be more precise) of the data that we want to use to exercise our test.
+It will then *generate* test cases that satisfy this description and will run our test on these cases.
+Ultimately this an extremely powerful tool for enabling us to write high-quality automated tests for our code.
+
+Hypothesis can be installed via conda:
+
+```shell
+conda install -c conda-forge hypothesis
```

Comment:
Hypothesis is [available from the default channel](https://anaconda.org/anaconda/hypothesis) as well as [conda-forge](https://anaconda.org/conda-forge/hypothesis).  The default channel is usually a week to a month behind, but since most `conda` users don't update regularly that probably doesn't matter much...  I'd just show the defaults install here.